src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/MigrateBookingStatusTest.kt

```diff
@@ -0,0 +1,321 @@
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val booking1 = createCAS3ArrivedBooking(userEntity)
+        val booking2 = createCAS3ArrivedBooking(userEntity)
+
+        assertBookingStatusIsNull(booking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.arrived, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.arrived, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate multiple CAS3 Confirmed booking and returns 202 response`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val booking1 = createCAS3ConfirmedBooking(userEntity)
+        val booking2 = createCAS3ConfirmedBooking(userEntity)
+
+        assertBookingStatusIsNull(booking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(booking2, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(booking1, BookingStatus.confirmed, ServiceName.temporaryAccommodation)
+        assertBookingStatus(booking2, BookingStatus.confirmed, ServiceName.temporaryAccommodation)
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully migrate CAS3 only 'departed' bookings when request is for 'departed' bookings and multiple booking exists in the DB`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val departedBooking1 = createCAS3DepartedBooking(userEntity)
+        val departedBooking2 = createCAS3DepartedBooking(userEntity)
+        val arrivedBooking = createCAS3ArrivedBooking(userEntity)
+        val confirmedBooking = createCAS3ConfirmedBooking(userEntity)
+        val provisionalBooking = createTemporaryAccommodationBooking(userEntity, null)
+
+        assertBookingStatusIsNull(departedBooking1, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(departedBooking2, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(arrivedBooking, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(confirmedBooking, ServiceName.temporaryAccommodation)
+        assertBookingStatusIsNull(provisionalBooking, ServiceName.temporaryAccommodation)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatus(departedBooking1, BookingStatus.departed, ServiceName.temporaryAccommodation)
+        assertBookingStatus(departedBooking2, BookingStatus.departed, ServiceName.temporaryAccommodation)
+        assertBookingStatus(arrivedBooking, BookingStatus.arrived, ServiceName.temporaryAccommodation)
```

Comment:
This test describes itself as

>  fun `Successfully migrate CAS3 only 'departed' bookings when request is for 'departed' bookings and multiple booking exists in the DB`() {

But I think it's actually updating the booking statuses for arrived, confirmed and provisional too. Perhaps the description is wrong or we need to make sure that these bookings still have a null status?

Reply:
Its my fault, i forget to change the description. Initially i developed an API to migrate 'status' by status and then realised we already got migration tool, so i changed the approach but forget to change the description.